hnn_core/network.py

```diff
@@ -525,25 +664,53 @@ def plot_spikes(self, ax=None, show=True):
         fig : instance of matplotlib Figure
             The matplotlib figure object
         """
+
         import matplotlib.pyplot as plt
-        spikes = np.array(sum(self.spiketimes, []))
-        gids = np.array(sum(self.spikegids, []))
-        spike_times = np.zeros((4, spikes.shape[0]))
+        spiketimes = np.array(sum(self.times, []))
+        spiketypes = np.array(sum(self.types, []))
         cell_types = ['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']
-        for idx, key in enumerate(cell_types):
-            mask = np.in1d(gids, self.gid_dict[key])
-            spike_times[idx, mask] = spikes[mask]
+        spiketimes_cell = [spiketimes[spiketypes == cell_type]
+                           for cell_type in cell_types]
 
         if ax is None:
             fig, ax = plt.subplots(1, 1)
 
-        ax.eventplot(spike_times, colors=['r', 'b', 'g', 'w'])
+        ax.eventplot(spiketimes_cell, colors=['r', 'b', 'g', 'w'])
         ax.legend(cell_types, ncol=2)
         ax.set_facecolor('k')
         ax.set_xlabel('Time (ms)')
         ax.get_yaxis().set_visible(False)
         ax.set_ylim((-1, 4.5))
+        ax.set_xlim(left=0)
 
         if show:
             plt.show()
         return ax.get_figure()
+
+    def write(self, fname):
+        """Write spiking activity to a collection of spike trial
+        files.
```

Comment:
This should be a single line. Quoting [PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings):

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.